src/hooks/useProducts.ts

```diff
@@ -0,0 +1,36 @@
+import { Product } from "@/types/products";
+import { useState } from "react";
+import { getProducts } from "@/apis/product";
+import { Category, Sort } from "@/constants/selectOption";
+import PRODUCT from "@/constants/product";
+import useMutation from "@/hooks/useMutation";
+import { ERROR_MESSAGES } from "@/constants/messages";
+
+const useProducts = () => {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [isLastPage, setIsLastPage] = useState(false);
+
+  const { mutate: getProductsMutate, isLoading } = useMutation<typeof getProducts>(
+    getProducts,
+    ERROR_MESSAGES.failGetProducts
+  );
+
+  const fetchProductPage = async (category: Category, page: number, sort: Sort) => {
+    const size = page === 0 ? PRODUCT.firstPageItemCount : PRODUCT.subsequentPageItemCount;
+
+    const res = await getProductsMutate({ category, page, size, sort });
+    if (!res) return;
+    if (res.last) setIsLastPage(true);
+
+    page === 0 ? setProducts(res.content) : setProducts((prevProducts) => [...prevProducts, ...res.content]);
```

Comment:
요기 코드의 가독성에 고민을 했는데 호프가 보기에 괜찮나요 ??

Reply:
아하 확실히 가독성이 조금 떨어지긴 하네요? 
아래 응답값에 따라서 lastPage 상태나, products 상태, currentPage 상태를 변경해주는 부분이 따로 함수로 분리되어 관리되면 더 좋겠네요 ㅎㅎ fetch 와는 엄연히 다른 역할을 하고 있으니까요?

Reply:
```suggestion
setProducts((prevProducts) => (page === 0 ? res.content : [...prevProducts, ...res.content]));
```

우선 해당 코드가 최선이라 생각하여 변경했습니다!